Sources/Stream.swift

```diff
@@ -32,13 +36,20 @@ extension Sending {
 }
 
 extension Receiving {
-    public func receive(upTo byteCount: Int, timingOut deadline: Double, completion: (Void throws -> Data) -> Void) {
+    public func receive(upTo byteCount: Int, timingOut deadline: Double, completion: (Void throws -> Received) -> Void) {
         completion { try self.receive(upTo: byteCount, timingOut: deadline) }
     }
 }
 
 extension Receiving {
-    public func receive(upTo byteCount: Int) throws -> Data {
+    public func receive(upTo byteCount: Int) throws -> Received {
         return try receive(upTo: byteCount, timingOut: .never)
     }
-}
\ No newline at end of file
+}
+
+public protocol BinaryStream: Stream {
+    associatedtype Interchanged = Data
```

Comment:
Never mind... It assumes the type but it does not make the stream itself concrete 😞